pkg/yaml/types.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2017 The Kedge Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package yaml
+
+import (
+	"encoding/json"
+	"errors"
+	"strconv"
+)
+
+// SliceArrayorIntArray represents a slice or an integer
+type SliceArrayorIntArray []string
```

Comment:
Could we have 

```go
type StringOrInt []string
type SliceArrayorIntArray []StringOrInt

func (s *StringOrInt) UnmarshalJSON(data []byte) error {
....
```

or something similar if possible,
so we can use it in #454 for `appversion` key that should have single value of IntOrString

Reply:
Do you mind if we add this when we implement #454 instead since it's unrelated to this current commit? Reason being is that each "key" needs it's own Unmarshaler and thus we would have to add a custom UnmarshalJSON for each "key" used in `types.go`, for example:

```golang

func (s *StringOrInt) UnmarshalJSON(data []byte) error {
....


func (s *SliceArrayorIntArray) UnmarshalJSON(data []byte) error {
....

etc.
```

Or I could open a new PR for this after this PR #455 is implemented for the new type.

Reply:
Another thing is that this it should be of type `int` instead of `string`. As the whole point of this is to parse string as an integer, so resulting value should be stored in `integer`

Reply:
@kadel Actually.. the whole thing is to parse back into a string. Hence why the result is []string

Reason being is that our PortMappings implemention by @containscafeine checks for "80:80/tcp" for example within the string. Hence the reason why all int's are converting to string and not the other-way around.

The end result would be something like this: []string{"80", "80:80/tcp", "3306:3306"} etc.

Having the output as a []int64 or []int wouldn't be possible..